test/pageobjects/page.js

```diff
@@ -1,6 +1,22 @@
+import config from "../pageconfig/config.json";
 export default class Page {
     // Global methods go here
-    open (path) {
-        browser.url(path)
+    open(path) {
+        browser.url(path);
+    }
+
+    waitPageToLoad() {
+        browser.waitUntil(
+            function () {
+                const state = browser.execute(function () {
```

Comment:
Since `browser.execute` is an async function, should this use `browser.executeAsync` to ensure that state is actually resolved before the assertion is returned?

Reply:
i think browser.executeAsync is async function while browser.execute is a sync function